packages/loader/container-definitions/src/loader.ts

```diff
@@ -112,6 +114,17 @@ export interface IContainer extends IEventProvider<IContainerEvents>, IFluidRout
      */
     readonly attachState: AttachState;
 
+    /**
+     * The current code details for the container's runtime
+     */
+    readonly codeDetails: IFluidCodeDetails | undefined
+
+    /**
+     * Propose new code details that define the code to be loaded
+     * for this container's runtime.
+     */
+    proposeCodeDetails(codeDetails: IFluidCodeDetails): Promise<void>
```

Comment:
The promise resolves once the proposal is accepted/rejected. It might be worth mentioning that in the comment as part of the contract.

Reply:
"Returns a promise that will resolve when the proposal is either accepted or rejected."
It only rejects if the proposal encounters an error.

Also it would be nice if it resolved with true for accept and false for reject, but out of scope probably.

Reply:
i like that idea, and not too difficult